presto-product-tests/src/main/java/io/prestosql/tests/cli/PrestoCliFileAuthTests.java

```diff
@@ -0,0 +1,190 @@
+        assertAuthenticationError();
+    }
+
+    @Test(groups = {FILEAUTH, PROFILE_SPECIFIC_TESTS}, timeOut = TIMEOUT)
+    public void testCorrectBCryptPassword()
+            throws IOException, InterruptedException
+    {
+        launchPrestoCliWithServerArgument("userBCrypt", "user123", "--execute", "select * from hive.default.nation;");
+        assertSuccess();
+    }
+
+    @Test(groups = {FILEAUTH, PROFILE_SPECIFIC_TESTS}, timeOut = TIMEOUT)
+    public void testIncorrectBCryptPassword()
+            throws IOException, InterruptedException
+    {
+        launchPrestoCliWithServerArgument("userBCrypt", "user1234", "--execute", "select * from hive.default.nation;");
+        assertAuthenticationError();
+    }
+
+    @Test(groups = {FILEAUTH, PROFILE_SPECIFIC_TESTS}, timeOut = TIMEOUT)
+    public void testAuthTokenCacheUpdate()
+            throws IOException, InterruptedException
+    {
+        launchPrestoCliWithServerArgument("userBCrypt", "user123", "--execute", "select * from hive.default.nation;");
+        assertSuccess();
+        //update the content in file and check for success
+        addContentToFile("/docker/volumes/conf/presto/etc/file_auth_copy", "userBCrypt1:$2y$10$BqTb8hScP5DfcpmHo5PeyugxHz5Ky/qf3wrpD7SNm8sWuA3VlGqsa");
+        Thread.sleep(10); // pausing till refresh-period for reloading the file
+        launchPrestoCliWithServerArgument("userBCrypt1", "user123", "--execute", "select * from hive.default.nation;");
+        assertSuccess();
+        //delete the content in file and check for failure
+        deleteContentToFile("/docker/volumes/conf/presto/etc/file_auth_copy", "userBCrypt1:$2y$10$BqTb8hScP5DfcpmHo5PeyugxHz5Ky/qf3wrpD7SNm8sWuA3VlGqsa");
+        launchPrestoCliWithServerArgument("userBCrypt1", "user123", "--execute", "select * from hive.default.nation;");
+        assertAuthenticationError();
+    }
+
+    @Test(groups = {FILEAUTH, PROFILE_SPECIFIC_TESTS}, timeOut = TIMEOUT)
+    public void testNonExistentUserFailure()
+            throws IOException, InterruptedException
+    {
+        launchPrestoCliWithServerArgument("userDNE", "passwordXYZ", "--execute", "select * from hive.default.nation;");
+        assertAuthenticationError();
+    }
+
+    private void assertSuccess()
+    {
+        assertThat(trimLines(presto.readRemainingOutputLines())).containsAll(nationTableBatchLines);
+        skipAfterTestWithContext();
+    }
+
+    private void assertAuthenticationError()
+    {
+        assertTrue(trimLines(presto.readRemainingErrorLines()).stream().anyMatch(str -> str.contains("Authentication failed")));
+        skipAfterTestWithContext();
+    }
+
+    private void skipAfterTestWithContext()
+    {
+        if (presto != null) {
+            presto.close();
+            presto = null;
+        }
+    }
+
+    private void addContentToFile(String filename, String content)
+    {
+        try {
```

Comment:
Use try-with-resources so that we clean up properly and throw on failure
```java
try (BufferedWriter writer = new BufferedWriter(new FileWriter(filename, true))) {
    writer.newLine();
    writer.write(content);
}

Reply:
Done